source/load-balancers/load-balancers.rst

```diff
@@ -0,0 +1,531 @@
+hello responses in order to tell the driver they were behind a load balancer.
+
+From a driver's view there is a change in behaviour in that once in load balanced mode, the driver
+can never transition out of that mode, unlike other states in the SDAM specification.
+
+Why does this specification instruct drivers to not check connections back into the connection pool in some circumstances?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the case of a load balancer fronting multiple services, it is possible that a connection to th
+e load balancer could result in a connection behind the load balancer to a different service. In
+order to guarantee these operations execute on the same service they need to be executed on the
+same socket - not checking a connection back into the pool for the entire operation guarantees this.
+
+What reason has a client side connection reaper for idle cursors not been put into this specification?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+It was discussed as a potential solution for maxed out connection pools that the drivers could
+potentially behave similar to the server and close long running cursors after a specified time
+period and return their connections to the pool. Due to the high complexity of that solution
+it was determined that better error messaging when the connection pool was maxed out would
+suffice in order for users to easily debug when the pool ran out of connections and fix their
+applications or adjust their pool options accordingly.
+
+Why are we requiring mongos servers to add a new serviceId field in hello responses rather than reusing the existing topologyVersion.processId?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This option was previously discussed, but we opted to add a new :code:`hello` response field in
+order to not mix intentions.
+
+Why does this specification not address load balancer restarts or maintenance?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+It was decided that the core problem that was needed to be solved was to introduce durable/resumable
+cursors to some services. The related ticket is `DRIVERS-1549 <https://jira.mongodb.org/browse/DRIVERS-1549>`__.
+
+
+Design Rationales
+-----------------
+
+This design is based on the simplest approach to implementation in both the drivers and all services,
+when taking into consideration that it is NOT a requirement that applications connected to any service
+should not need to make changes and/or restart their applications. If this requirement were to change,
+see alternative designs below.
+
+Connection pinning behaviour is not altered in this specification and it is understood that this can
```

Comment:
I'm not sure I understand this paragraph. It says that connection pinning behavior is not altered in this spec, but I'm not aware of any previous features that required drivers to pin connections in the first place. Also, the wording about session and cursor leaks reads as if the driver is intentionally leaking resources, which is not true. Overall, I'm not sure this paragraph adds much value so maybe we can simply remove it?

Reply:
I've removed this paragraph now.